ships/0024-surfacing-errors-to-buildrun.md

```diff
@@ -117,11 +117,56 @@ As a third-party application developer that uses shipwright under the hood, I wa
 
 ### Implementation Notes
 
-`tbd in the next phase of the proposal`
+
+This ship proposes two changes in the API that are implemented in `buildrun_types.go` and thus propagated in the CRD. First, the BuildRun's status gets the new field `errorDetails` that embeds the `failedAt` field:
+
+```go
+type FailureDetails struct {
+    Reason   string    `json:"reason,omitempty"`
+    Message  string    `json:"message,omitempty"`
```

Comment:
Would it make sense to place `Reason` and `Message` in the `Succeeded` status condition?

I generally agree that we should surface additional context on failure. My concern is that having reasons/message here that differ from the `Succeeded` status condition may confuse users. And if we need to present the same information, there is no sense in duplication.

What if we had something like this?

```yaml
status:
  conditions:
  - type: Succeeded
    status: False
    reason: GitBranchNotFound
    message: "Git branch my-feature not found"
  failure:
    pod: my-buildrun-d29sx
    container: step-git-clone
```